src/ngx_http_lua_api.c

```diff
@@ -194,13 +198,12 @@ ngx_http_lua_shared_memory_init(ngx_shm_zone_t *shm_zone, void *data)
 
     lmcf->shm_zones_inited++;
 
-    if (lmcf->shm_zones_inited == lmcf->shm_zones->nelts
-        && lmcf->init_handler)
+    if (lmcf->shm_zones_inited == lmcf->shm_zones->nelts)
```

Comment:
@agentzh because I think use like a third module here: they can not get `lmcf->init_handler`,  and we can check init handler in `ngx_http_lua_delay_init_phase` api